jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudformation/CloudFormationTemplate.kt

```diff
@@ -37,11 +38,31 @@ interface CloudFormationTemplate {
     }
 }
 
-interface Resource {
+interface NamedMap {
     val logicalName: String
 
-    fun isType(requestedType: String): Boolean
-    fun type(): String?
     fun getScalarProperty(key: String): String
+    fun getOptionalScalarProperty(key: String): String?
     fun setScalarProperty(key: String, value: String)
-}
\ No newline at end of file
+}
+
+interface Resource : NamedMap {
+
```

Comment:
Nit: extra space

Reply:
I initially submitted without the space, and the build failed on the lint rules

Reply:
The blank line? (line 50) is what kyle is saying i think